clang/lib/CConv/ConstraintBuilder.cpp

```diff
@@ -311,13 +311,17 @@ class GlobalVisitor : public RecursiveASTVisitor<GlobalVisitor> {
       // If the location of the previous RecordDecl and the current VarDecl are the same,
       // this implies an inline struct as per Clang's AST, so set a flag in ProgramInfo
       // to indicate that this variable should be constrained to wild later
-      if(G->getBeginLoc().getRawEncoding() == LastRecordLoc) { Info.InlineStructEncountered = true; }
       Info.addVariable(G, Context);
       if (G->hasInit()) {
         CB.constrainLocalAssign(nullptr, G, G->getInit());
       }
-      // For safety, reset the flag to false after the variable has been added
-      Info.InlineStructEncountered = false;
+      if(InLineStructEncountered) {
+        const std::set<ConstraintVariable *> &C = Info.getVariable(G, Context);
+        for(const auto &Var : C) {
+          std::string Rsn = "Variable " + G->getNameAsString() + " is an inline struct definition.";
+          Var->constrainToWild(Info.getConstraints(), Rsn);
+        }
+      }
     }
```

Comment:
Don't you need to reset the `InlineStructEncountered` flag once you've processed definitions that use it?

Reply:
I do! Thanks for pointing this out. I'll fix it now.

Reply:
For test cases: I would want to confirm that it works with
```
// a, b, c below all stay as WILD pointers
struct foo { int *x; } *a; // one decl; x rewrites to _Ptr<int>
struct bar { int* y; } *b, *c; // two decls, not one; y stays as int *
void foo(void) {
  a->x = (void*)0; 
  b->y = (void*)0;
  c->y = (int *)5; // forces it to WILD
}
```
You might also try a test where the `struct` is defined within a function, rather than globally.

Reply:
Ok fixed!

Reply:
Are you sure this code is going to work with `struct bar { int* y; } *b, *c;`? Notice that there are two variables being declared here, not one. But it would seem that you are resetting the `InLineStructEncountered` eagerly, perhaps before the second variable is dealt with?

Also, I am not seeing a change to `VisitVarDecl`, i.e., for global variables. Did I miss it?

Reply:
Your test cases don't seem to have two variables being processed, at least not here.

You seem to also not have the case that there's a non-inline struct declared in between. This is what my test case was trying to show.

Reply:
You're right in that the *c will actually get converted to a _Ptr. This might have been a misinterpretation on my part, but I think in a meeting you said the struct should only ever be defined once. So in the inline struct definition for b, struct bar is defined, and then when we come to c (which has now been split), there is no need to redefine struct bar, so c is allowed to be a _Ptr. It's clearly not consistent, so I understand the cause for concern. In fact, I'm just noting in your comment above that you expect C to stay WILD. Should I add support for C to become wild as well?

Reply:
Oh, I didn't realize that `c` was going to be split to its own definition. Interesting. Yes, for consistency (and to avoid confusion) I think it's better if they are all WILD, not just some of them.

Reply:
Gotcha. I don't understand what you mean by no change to VisitVarDecl. I think I call the function processInLineStruct for global vars. Is it not there on your end?

Reply:
That was another private static method I created because the constrain to wild occurs in both the function and global inline structs, so I abstracted it away into that function. Is that ok?

Reply:
I see. This is good -- thanks!